.circleci/config.yml

```diff
@@ -12,8 +12,39 @@ jobs:
       - run:
           name: Build
           command: ./gradlew check build assembleAndroidTest
+  publish_to_maven:
+    docker:
+      - image: circleci/android:api-30
+    steps:
+      - run:
+          name: Publish to Maven
+          command: ./gradlew publish
+  publish_github_release:
+    docker:
+      - image: cibuilds/github:0.10
+    steps:
+      - run:
+          name: Publish Release on GitHub
+          command: ./gradlew createGitHubRelease
 workflows:
-  version: 2
-  build_and_test:
+  build_test_and_deploy:
     jobs:
       - build_and_test
+      - publish_to_maven:
+          context: Global
+          filters:
+            branches:
+              only:
+                - "master"
+                - "develop"
```

Comment:
does the `develop` branch create an internal release?

Reply:
does this mean our PRs should always be targeting `develop` instead of `master`?

Reply:
I'm leaning towards having separate branch for maven / github releases, because this would help us explicitly control version upgrades and reduce outstanding PRs.

For e.g.

We have 1000 PRs from 100s of happy developers. Typically they'd target `master` as that's the most convenitonal way of opening a PR. 

We don't want 1000 separate releases after merging, especially because we'd have to manage bumping up the version each time.

Instead, if we had a locked down `production` branch, then we could publish releases on our own schedule with more intentional versioning.

Reply:
Maybe cc @nickpatrick  / @kochis / @tjulien to see what their thoughts are.

Reply:
That being said, GitHub and Maven support "staging" / draft publishing, so we could adopt the same convention as we do with server and FE, which is:

```
master = staging
production = production
```

In the release process, it could look like:

```
master = draft publish + overwrite current release
production = finalized publish, generally don't want to overwrite releases
```

Reply:
@jkao let's take this discussion offline / move to slack